### PR TITLE
ES6/7 refactor, Remove jQuery

### PR DESCRIPTION
**Pull request by luigiplr**
PR will:

-Bring codebase up to ES6 spec while utilizing a few select ES7 features. #syntaxSugar 
-Remove jQuery as a dep altogether

### CODE REVIEW

File: app/scripts/views/table.js [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-    children: React.PropTypes.array
-  },
-  render: function () {
-    return (
-      <Table responsive>
+
+export
+default class Table extends React.Component {
+
+    static propTypes: {
```
Comment (Dignifiedquire) [inline app/scripts/views/table.js]: this needs to be an `=` instead of `:` I think

File: app/scripts/app.js [hunk 2]
```diff
@@ -1,17 +1,14 @@
 import React from 'react'
-import {render} from 'react-dom'
-import Router, {hashHistory} from 'react-router'
```
Comment (Dignifiedquire) [inline app/scripts/app.js]: Why did you change this import?

File: app/scripts/include/globe.js [hunk 3]
```diff
@@ -16,14 +16,16 @@ import debug from 'debug'
 
 const log = debug('globe:scripts')
 
-export default function (container, opts) {
+export
+default
+function(container, opts) {
```
Comment (Dignifiedquire) [inline app/scripts/include/globe.js]: why is this spread over three lines?

File: app/scripts/app.js (hunk 2 above)
Comment (luigiplr) [inline app/scripts/app.js]: Standard format screamed.

File: app/scripts/include/globe.js [hunk 4]
```diff
@@ -16,14 +16,16 @@ import debug from 'debug'
...
   opts = opts || {}
 
-  var colorFn = opts.colorFn || function (x) {
-      var c = new THREE.Color()
-      c.setHSL((0.6 - (x * 0.5)), 1.0, 0.5)
-      return c
-    }
+  var colorFn = opts.colorFn || function(x) {
```
Comment (Dignifiedquire) [inline app/scripts/include/globe.js]: I thought we are dropping var?

File: app/scripts/include/globe.js [hunk 5]
```diff
@@ -108,7 +110,7 @@ export default function (container, opts) {
   var padding = 40
   var PI_HALF = Math.PI / 2
 
-  function init () {
+  function init() {
```
Comment (Dignifiedquire) [inline app/scripts/include/globe.js]: please revert, standard willl not be happy with this

Comment (luigiplr) [inline app/scripts/include/globe.js]: Done.

File: app/scripts/pages/files.js [hunk 6]
```diff
@@ -11,250 +16,243 @@ import i18n from '../utils/i18n.js'
...
+            pinned: [],
+            local: [],
+            dragging: false
+        }
 
-      t.props.ipfs.pin.list('recursive', function (err, pinned) {
-        if (err || !pinned) return t.error(err)
-        t.setState({ local: Object.keys(pinned.Keys).sort() })
-      })
+        this._renderTitle = this._renderTitle.bind(this)
```
Comment (Dignifiedquire) [inline app/scripts/pages/files.js]: Lets not do this but rather use 


_renderTitle = () => {
  //....
}

Comment (luigiplr) [inline app/scripts/pages/files.js]: Alrighty

File: app/scripts/include/globe.js (hunk 3 above)
Comment (Dignifiedquire) [inline app/scripts/include/globe.js]: ping

Comment (luigiplr) [inline app/scripts/include/globe.js]: Fixing

File: app/scripts/pages/bitswap.js [hunk 7]
```diff
@@ -1,44 +1,51 @@
...
-    t.props.pollInterval = setInterval(update, 1000)
-
-    return {
+import {
+  Row, Col, Panel
+}
+from 'react-bootstrap'
+
+export
+default class extends React.Component {
```
Comment (Dignifiedquire) [inline app/scripts/pages/bitswap.js]: Lets give the classes names as well `default class Bitswap extends React.Component`

File: app/scripts/pages/config.js [hunk 8]
```diff
@@ -15,14 +20,10 @@ export default React.createClass({
...
 
-    return { config: null }
-  },
-
-  render: function () {
-    var config = this.state.config
-      ? <ConfigView config={this.state.config} ipfs={this.props.ipfs} />
-      : null
+  render () {
+    const config = this.state.config ? <ConfigView config={this.state.config} ipfs={this.props.ipfs} /> : null
```
Comment (Dignifiedquire) [inline app/scripts/pages/config.js]: Can we change this to an early return, makes it much easier to read

```js
if (!this.state.config) return null

return <ConfigView />
```

File: app/scripts/pages/connections.js [hunk 9]
```diff
@@ -1,84 +1,95 @@
...
-  propTypes: {
-    peers: React.PropTypes.array
-  },
-  getInitialState: function () {
-    return {
+export
+default class extends React.Component {
+  constructor () {
+    super()
+    this.state = {
```
Comment (Dignifiedquire) [inline app/scripts/pages/connections.js]: Let's use ES7ish style initializers for state:

```js
class Connections extends Component {
}
```

File: app/scripts/pages/bitswap.js (hunk 7 above)
Comment (luigiplr) [inline app/scripts/pages/bitswap.js]: Sure

File: app/scripts/pages/connections.js [hunk 10]
```diff
@@ -1,84 +1,95 @@
...
+  static displayName = 'Connections';
+  static propTypes = {
+    peers: React.PropTypes.array,
+    ipfs: React.PropTypes.object
+  };
 
-  componentDidMount: function () {
-    var t = this
+  componentDidMount () {
+    this.mounted = true
```
Comment (Dignifiedquire) [inline app/scripts/pages/connections.js]: This should be `this.setState({mounted: true})`

File: app/scripts/pages/files.js [hunk 11]
```diff
@@ -142,16 +127,16 @@ export default React.createClass({
...
-  _renderAdder () {
+  _renderAdder = () => {
     return (
       <div className='file-add-container'>
         <div
-            className='file-add-target'
-            onDragOver={this.onDragOver}
-            onDragLeave={this.onDragLeave}
-            onDrop={this.onDrop}
+          className={'file-add-target ' + (this.state.dragging ? 'hover' : null)}
```
Comment (Dignifiedquire) [inline app/scripts/pages/files.js]: this really should use [classnames](https://www.npmjs.com/package/classnames)

File: app/scripts/pages/logs.js [hunk 12]
```diff
@@ -65,40 +80,30 @@ export default class Logs extends Component {
...
-  }
-
   render () {
     const buttons = (
       <div className='buttons'>
-        <button className='btn btn-second' onClick={this.clear}>{i18n.t('Clear')}</button>
+        <button className='btn btn-second' onClick={() => this.setState({ log: [] })}>{i18n.t('Clear')}</button>
         <button className={'btn btn-second ' + (this.state.tailing ? 'active' : '')}
-          data-toggle='button' aria-pressed={this.state.tailing} onClick={this.toggleTail}>{i18n.t('Tail')}</button>
+          data-toggle='button' aria-pressed={this.state.tailing} onClick={() => this.setState({ tailing: !this.state.tailing })}>{i18n.t('Tail')}</button>
```
Comment (Dignifiedquire) [inline app/scripts/pages/logs.js]: Please break up this line, this is getting very long and hard to read

File: app/scripts/pages/logs.js [hunk 13]
```diff
@@ -65,40 +80,30 @@ export default class Logs extends Component {
...
-
-  toggleTail () {
-    this.setState({ tailing: !this.state.tailing })
-  }
-
   render () {
     const buttons = (
       <div className='buttons'>
-        <button className='btn btn-second' onClick={this.clear}>{i18n.t('Clear')}</button>
+        <button className='btn btn-second' onClick={() => this.setState({ log: [] })}>{i18n.t('Clear')}</button>
```
Comment (Dignifiedquire) [inline app/scripts/pages/logs.js]: Please break up this line, this is getting very long and hard to read

File: app/scripts/pages/objects.js [hunk 14]
```diff
@@ -77,30 +93,26 @@ export default React.createClass({
...
-    var error = this.state.error
-      ? <div className='row'>
-          <h4>{i18n.t('Error')}</h4>
-          <div className='panel panel-default padded'>
-            {this.state.error.Message}
-          </div>
+  }
+
+  render () {
+    const error = this.state.error ? (<div className='row'>
```
Comment (Dignifiedquire) [inline app/scripts/pages/objects.js]: Please use `if` for multiline conditions

File: app/scripts/pages/objects.js [hunk 15]
```diff
@@ -109,11 +121,11 @@ export default React.createClass({
             <h4>{i18n.t('Enter hash or path')}</h4>
             <Row className='path'>
               <Col xs={10}>
-                <input type='text' className='form-control input-lg' onChange={this.updatePath} onKeyPress={this.update} value={this.state.pathInput} placeholder={i18n.t('Enter hash or path: /ipfs/QmBpath...')}/>
+                <input type='text' className='form-control input-lg' onChange={event => this.setState({pathInput: event.target.value.trim()})} onKeyPress={this._update.bind(this)} value={this.state.pathInput} placeholder={i18n.t('Enter hash or path: /ipfs/QmBpath...')}/>
```
Comment (Dignifiedquire) [inline app/scripts/pages/objects.js]: Please break up this line, this is getting very long and hard to read

File: app/scripts/pages/objects.js [hunk 16]
```diff
@@ -109,11 +121,11 @@ export default React.createClass({
...
               </Col>
               <Col xs={2}>
                 <Button bsStyle={'primary'} className={'go'}
-                        onClick={this.update}>
+                        onClick={this._update.bind(this)}>
```
Comment (Dignifiedquire) [inline app/scripts/pages/objects.js]: You can use the `update = () => {};` trick again instead of binding manually

File: app/scripts/pages/routing.js [hunk 17]
```diff
@@ -1,43 +1,52 @@
...
+let base58Index = {}
+for (let i = 0; i < base58Chars.length; i++) {
   base58Index[base58Chars[i]] = i / 58
 }
 
-function idToAngle (id) {
-  var sub = id.substr(2, 4)
-  var angle = 0
-  for (var i = 0; i < sub.length; i++) {
+const idToAngle = id => {
```
Comment (Dignifiedquire) [inline app/scripts/pages/routing.js]: This really belongs into the `helpers` directory

File: app/scripts/pages/routing.js [hunk 18]
```diff
@@ -45,22 +54,20 @@ export default React.createClass({
...
-    }
-  },
-
-  render: function () {
+  render () {
     return (
       <Row>
         <Col sm={10} smOffset={1}>
-          <DHTGraph peers={this.state.peers}/>
+          <DHTGraph peers={this.state.peers}/ >
```
Comment (Dignifiedquire) [inline app/scripts/pages/routing.js]: unneeded whitespace

File: app/scripts/pages/routing.js [hunk 19]
```diff
@@ -45,22 +54,20 @@ export default React.createClass({
...
         </Col>
-      </Row>
+      </Row >
```
Comment (Dignifiedquire) [inline app/scripts/pages/routing.js]: unneeded whitespace

File: app/scripts/utils/localStorage.js [hunk 20]
```diff
@@ -5,32 +5,31 @@ const log = debug('utils:localStorage')
...
   // Allow setting objects directly
   if (op === 'setItem' && args[1] !== 'string') {
     args[1] = JSON.stringify(args[1])
   }
-
   try {
-    var result = ls[op].apply(ls, args)
-    // If getting, attempt to parse
+    let result = ls[op].apply(ls, args)
+      // If getting, attempt to parse
```
Comment (Dignifiedquire) [inline app/scripts/utils/localStorage.js]: wrong indent

File: app/scripts/views/connection.js [hunk 21]
```diff
@@ -1,47 +1,43 @@
...
-        location={this.props.location}
-        bytesRead={this.props.BytesRead}
-        bytesWritten={this.props.BytesWritten} />
-      : null
+  }
 
-    var className = 'webui-connection list-group-item'
-    if (this.state.open) className += ' active'
+  render () {
+    const peer = this.state.open ? <Peer
```
Comment (Dignifiedquire) [inline app/scripts/views/connection.js]: please use an `if` statement

File: app/scripts/views/dhtgraph.js [hunk 22]
```diff
@@ -1,55 +1,64 @@
...
+
+export
+default class extends React.Component {
+  constructor () {
+    super()
+
+    this.state = {
+      initialized: false
+    }
+    this.update = this.update.bind(this)
```
Comment (Dignifiedquire) [inline app/scripts/views/dhtgraph.js]: not needed if `update` is defined as ` update = () => {}`

File: app/scripts/views/filelist.js [hunk 23]
```diff
@@ -41,43 +35,40 @@ export default React.createClass({
...
-                    <i className='fa fa-remove'></i>
-                  </OverlayTrigger>
-                </a>
-              </td>
-            </tr>
-          )
-        }) : void 0}
         </tbody>
-      </Table>
+      </Table >
```
Comment (Dignifiedquire) [inline app/scripts/views/filelist.js]: unneeded whitespace

File: app/scripts/pages/connections.js (hunk 9 above)
Comment (luigiplr) [inline app/scripts/pages/connections.js]: np

File: app/scripts/views/globe.js [hunk 24]
```diff
@@ -8,84 +7,79 @@ require('../include/Detector.js')
...
-  },
-
-  addPoints: function () {
-    var data = this.parsePeers()
+  _addPoints () {
+    const data = this.parsePeers()
     if (!this.globe || !data.length) return
-    // TODO find difference between old points and new points
-    // and only add the new ones. THREE might be doing this internally.
+      // TODO find difference between old points and new points
```
Comment (Dignifiedquire) [inline app/scripts/views/globe.js]: wrong indentation

File: app/scripts/pages/config.js (hunk 8 above)
Comment (luigiplr) [inline app/scripts/pages/config.js]: Righto

File: app/scripts/views/page.js [hunk 25]
```diff
@@ -1,156 +1,140 @@
...
+                                </a>
+                            </li>
+                            <li>
+                                <a href='https://github.com/ipfs/webui' target='_blank' data-toggle='tooltip' data-placement='bottom' title={i18n.t('Github Repository')}>
+                                    <img src={require('../../img/git.png')} alt='Github' className='img-responsive icon'/><span className='sr-only'>{i18n.t('Github')}</span>
+                                </a>
+                            </li>
+                            <li>
+                                <a href='https://github.com/ipfs/webui/issues/new' target='_blank' data-toggle='tooltip' data-placement='bottom' title={i18n.t('Report Bugs')}>
+                                    <img src={require('../../img/bug.png')} alt='Report a bug' className='img-responsive icon'/><span className='sr-only'>{i18n.t('Report a bug')}</span>
```
Comment (Dignifiedquire) [inline app/scripts/views/page.js]: please break up this line

File: app/scripts/views/page.js [hunk 26]
```diff
@@ -1,156 +1,140 @@
...
+                        </form>
+                          <ul className='nav navbar-nav navbar-right collapse navbar-collapse'>
+                            <li>
+                                <a href='http://ipfs.io' target='_blank' data-toggle='tooltip' data-placement='bottom' title={i18n.t('About IPFS')}>
+                                    <img src={require('../../img/help.png')} alt='Help' className='img-responsive icon'/><span className='sr-only'>{i18n.t('Help')}</span>
+                                </a>
+                            </li>
+                            <li>
+                                <a href='https://github.com/ipfs/webui' target='_blank' data-toggle='tooltip' data-placement='bottom' title={i18n.t('Github Repository')}>
+                                    <img src={require('../../img/git.png')} alt='Github' className='img-responsive icon'/><span className='sr-only'>{i18n.t('Github')}</span>
```
Comment (Dignifiedquire) [inline app/scripts/views/page.js]: please break up this line

File: app/scripts/views/page.js [hunk 27]
```diff
@@ -1,156 +1,140 @@
...
-            {update}
-            {this.props.children && React.cloneElement(
-               this.props.children, {ipfs: ipfs, host: ipfsHost, gateway: this.state.gateway}
-            )}
+                          <button className='btn btn-third btn-xs' onClick={() => this.context.router.push(`/objects/${parse(this.refs['dag-path'].val()).urlify()}`)}>{i18n.t('GO')}</button>
+                        </form>
+                          <ul className='nav navbar-nav navbar-right collapse navbar-collapse'>
+                            <li>
+                                <a href='http://ipfs.io' target='_blank' data-toggle='tooltip' data-placement='bottom' title={i18n.t('About IPFS')}>
+                                    <img src={require('../../img/help.png')} alt='Help' className='img-responsive icon'/><span className='sr-only'>{i18n.t('Help')}</span>
```
Comment (Dignifiedquire) [inline app/scripts/views/page.js]: please break up this line

File: app/scripts/pages/connections.js (hunk 10 above)
Comment (luigiplr) [inline app/scripts/pages/connections.js]: This however would prevent it from being set to false on componentWIllUnmount.

**Comment by Dignifiedquire** [comment]
Getting close :) just 9 linting errors and one test not passing: https://circleci.com/gh/ipfs/webui/273

**Comment by Dignifiedquire** [comment]
Looking good, just missing the test failure (you can run it locally with `npm test`)

**Comment by Dignifiedquire** [comment]
Also needs merge with latest master please :)

**Comment by Dignifiedquire** [comment]
@luigiplr one last thing, can you squash your commits into one, and merge with latest master one more time. Then I can merge it :octocat:

**Comment by luigiplr** [comment]
@Dignifiedquire Sure thing

**Comment by Dignifiedquire** [comment]
Merged with master and did some cleanup in #216. Thanks @luigiplr for all the work!
